docs/source/config.rst

```diff
@@ -0,0 +1,117 @@
+
+************************
+About Configuration File
+************************
+
+Configuration files are `YAML <https://yaml.org/>`__ files. Under the hood,
+CMinx relies on the
+`Confuse <https://confit.readthedocs.io/en/latest/#>`__ library for locating,
+loading, and managing the configuration file. Users can directly provide
+CMinx with the path to their configuration file or place their configuration
+file in one of the paths searched by Confuse (*e.g.*, ``/etc/cminx/`` or
+``~/.config/cminx/``). See
+`here <https://confuse.readthedocs.io/en/latest/usage.html#search-paths>`__ for
+more details regarding how Confuse locates configuration files.
+
+For organizational and namespacing purposes we have partitioned the
+options recognized by CMinx into four categories:
+
+1. ``input``.   Affects which files and file parts CMinx will parse.
+2. ``output``.  Affects how CMinx outputs results.
+3. ``logging``. Options pertaining to CMinx's log messages. See
+   :ref:`logging_options` for more details.
+4. ``rst``.     Used to control the formatting of the generated reST.
+
+****************************
+Writing a Configuration File
+****************************
+
+One of the first things CMinx does when it is run is to load the default
+values for all options. If a user provides a configuration file, the
+configuration file will be used to override initially loaded default values.
+Hence, user configuration files only need to include options with non-default
+values (listing options with defaulted values will not hurt anything though).
+For example, by default CMinx will generate API documentation for all functions,
+regardless of whether they are documented or not (if they are not documented
+the generated documentation is little more than the signature). Let's say we
+do not want CMinx to generate documentation for undocumented functions. In this
+case, we create a file ``config.yaml`` and put the following in it:
+
+.. code-block:: yaml
+
+   input:
+     include_undocumented_function: false
+
+Assuming ``config.yaml`` lives in the directory we are invoking CMinx from, we
+then run:
+
+.. code-block:: terminal
+
+   cminx -s config.yaml <other options>
+
+If the user is calling CMinx from Python then the configuration file is passed
+directly to the ``main`` function like:
+
+.. code-block:: Python
+
+   import cminx
+   cminx.main(['-s', 'config.yaml', <other options>])
+
+(in both code examples ``<other options>`` should be replaced with any
+additional command line options the user wants to pass to CMinx).
+
+
+.. _logging_options:
+
+*******************
+Logging Options
+*******************
+In CMinx's configuration files, there is a `logging` section. This section is
+special in that it forwards all options inside of it to the Python
+`logging.config` module. This means that users can configure the Python logging
+ system, and thus CMinx's logging settings, directly from a standard config
+ file.
```

Comment:
There appears to be extra indentations on these lines that cause the generated HTML to look a bit funny